modules/task/execution.go

```diff
@@ -37,14 +37,14 @@ type fnRegister struct {
 
 var fnLookup = fnRegister{fnNameMap: make(map[string]interface{})}
 
-// fnSignature represents a function and its arguments
-type fnSignature struct {
+// FnSignature represents a function and its arguments
+type FnSignature struct {
```

Comment:
Can we expose it with a New()... function returning a Backend interface, not a exporting a struct?